### DISCUSSION TITLE
how to import `ast.Keyword`

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1532] [Q&A] [answered]
for langium 3.0,i define a class extend DefaultCompletionProvider and override completionForKeyword,but how to import `ast`
![image](https://github.com/eclipse-langium/langium/assets/26995040/629d91d3-1f0f-4ddb-8ece-41d2d357454b)

**Comment by msujew** [comment, marked as answer]
All grammar elements are exported via the `GrammarAST` object in `langium`.

**Reply by zrwsmd** [reply]
so it should like this ,right?
![image](https://github.com/eclipse-langium/langium/assets/26995040/5a0bebc1-160b-4447-b7c8-eba2dd51229b)

but i override this method in my test project(which langium version is 1.0.1)and code works out well,but when i move same code to my real project(langium 3.0),it seems didn't work out ,a little weird

**Reply by msujew** [reply]
Yes, that's the correct way to use it. In what way does it break in v3? Can you give more context?

**Reply by zrwsmd** [reply]
![image](https://github.com/eclipse-langium/langium/assets/26995040/89555aec-07c1-4a75-a349-851c763e4485)

**Reply by msujew** [reply]
Oh, ok. So it compiles, but doesn't yield the result you expect. I was a bit confused because the original question was about an import issue.

We've had a few changes to the completion logic in v2 of Langium, so there might've been a breaking change in the behavior of the completion. You'll need to debug into the completion provider to see what goes wrong. Remote diagnosis will be pretty difficult.

**Reply by zrwsmd** [reply]
ok,i get it,it because v3 project can't debug into  ,that is why i use old langium project which can debug into,i try to fix it

**Comment by zrwsmd** [comment]
keywordStr: string[] = [
        'SINT',
        'INT',
        'DINT',
        'LINT',
        'USINT',
        'UINT',
        'UDINT',
        'ULINT',
        'BYTE',
        'WORD',
        'DWORD',
        'LWORD',
        'TIME',
        'DATE',
        'LREAL',
        'REAL',
        'BOOL',
        'STRING'
    ];


   matchStrings(input: string): string[] {
        const regex = new RegExp('^' + input, 'i'); // 创建不区分大小写的正则表达式
        return this.keywordStr.filter(str => regex.test(str)); // 使用filter方法根据正则表达式进行匹配
    }
